### PR TITLE
lshw: B.02.18 -> 02.19

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Description of changes

- Bump version
- Switch to `fetchFromGitHub` to get updates from @r-ryantm bot
- Remove patches (they're merged upstream)
- Update dependencies
- Fix version workaround (not necessary anymore)
- Add myself as maintainer
- Fix the version for repology.org (see commit message for details)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by superherointj — APPROVED** [review]
`lshw` (CLI) runs for me. GUI not tested.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Review by thiagokokada — COMMENTED** [review]
Sorry, just realized that the `lshw -version` would report the wrong version. Now this is fixed.

File: pkgs/tools/system/lshw/default.nix [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
-  # Fix version info.
-  preConfigure = ''
-    sed -e "s/return \"unknown\"/return \"${version}\"/" \
-        -i src/core/version.cc
-  '';
-
-  makeFlags = [ "PREFIX=$(out)" ];
+  makeFlags = [
+    "PREFIX=$(out)"
+    "VERSION=${actualVersion}"
```
Comment (SuperSandro2000) [inline pkgs/tools/system/lshw/default.nix:36]:
```suggestion
    "VERSION=${src.rev}"
```

File: pkgs/tools/system/lshw/default.nix [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+, sqlite # compile GUI
+, withGUI ? false
 }:
 
+let
+  # Fix repology.org by not including the prefixed B, otherwise the `pname` attr
+  # gets filled as `lshw-B.XX.XX` in `nix-env --query --available --attr nixpkgs.lshw --meta`
+  # See https://github.com/NixOS/nix/pull/4463 for a definitive fix
+  version = "02.19";
+  actualVersion = "B.02.19";
```
Comment (SuperSandro2000) [inline pkgs/tools/system/lshw/default.nix:16]:
```suggestion
  version = "02.19";
  actualVersion = "B.${version}";
```
we don't want to duplicate this but we also don't need this here.

Comment (thiagokokada) [reply pkgs/tools/system/lshw/default.nix]: Done.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**